gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -335,7 +335,9 @@ public AddSpecResponse onAddSpec(Spec addedSpec) {
       if (quotaManager.isPresent()) {
         // QuotaManager has idempotent checks for a dagNode, so this check won't double add quotas for a flow in the DagManager
         try {
+          // todo : we should probably check quota for all of the start nodes
           quotaManager.get().checkQuota(dag.getNodes().get(0));
+          ((FlowSpec) addedSpec).getConfigAsProperties().setProperty(ServiceConfigKeys.GOBBLIN_SERVICE_ADHOC_FLOW, "true");
```

Comment:
Would it make sense if we added this in `createFlowSpecForConfig()` under `FlowConfigLocalResourceHandler.java`? Seems like this property can be used in a lot of areas outside of just this feature

Reply:
that function notifies the `addSpec` immediately after so I'm not sure we will use the value directly anywhere else. However, if we do use the in memory value of the Spec in other places, this property will be useful to add there. Btw shouldn't this value be set to false after the first iteration of an adhoc flow in the `DagManager` or Scheduler?

Reply:
Actually, the FlowSpec created by `createFlowSpecForConfig` will be persisted in the spec store. If we add the config there, it will be persisted permanently in the spec store. And then we will not be able to distinguish if it is a scheduled run or the adhoc run of a "schedule with runImmediately=true" kind of flows.
cc @ZihanLi58

Reply:
@umustafi the if block inside which this statement is present, will be false for the second iteration.
   ` !jobConfig.containsKey(ConfigurationKeys.JOB_SCHEDULE_KEY) || PropertiesUtils.getPropAsBoolean(jobConfig, ConfigurationKeys.FLOW_RUN_IMMEDIATELY, "false"))
`
run immediately is set to false on service restart and if job schedule key is absent , the flow would be deleted from the spec store after the 1st iteration.

I will add a comment there explaining this.